research/gam/gam/trainer/trainer_cotrain.py

```diff
@@ -355,12 +355,15 @@ def _select_samples_to_label(self, data, trainer_cls, session):
         assign to each of the selected nodes.
     """
     # Select the candidate samples for self-labeling, and make predictions.
-    # Remove the validation samples from the unlabeled data, if there, to avoid
-    # self-labeling them.
+    # Remove the validation and test samples from the unlabeled data, if there,
+    # to avoid self-labeling them. We could potentially leave the test edges
+    # but once a node self-labeled, is label is fixed for the remaining 
```

Comment:
Looks like this line has typos.

Reply:
Thanks for catching this! I've fixed it.